core/templates/pages/story-editor-page/editor-tab/story-editor.component.spec.ts

```diff
@@ -0,0 +1,532 @@
+      title: 'Title 3',
+      description: 'Description 3',
+      prerequisite_skill_ids: ['skill_1'],
+      acquired_skill_ids: ['skill_2'],
+      destination_node_ids: ['node_2'],
+      outline: 'Outline',
+      exploration_id: null,
+      outline_is_finalized: false,
+      thumbnail_bg_color: '#a33f40'
+    })];
+
+    const event1 = {
+      previousIndex: 0,
+      currentIndex: 1,
+    } as CdkDragDrop<string[]>;
+    const event2 = {
+      previousIndex: 1,
+      currentIndex: 0,
+    } as CdkDragDrop<string[]>;
+
+    component.drop(event1);
+    tick();
+    component.drop(event2);
+    tick();
+
+    expect(storyUpdateService.rearrangeNodeInStory).toHaveBeenCalled();
+  }));
+
+  it('should init the controller', () => {
+    expect(component.storyPreviewCardIsShown).toEqual(false);
+    expect(component.mainStoryCardIsShown).toEqual(true);
+    expect(component.getTopicName()).toEqual('addition');
+  });
+
+  it('should toggle story preview card', () => {
+    component.storyPreviewCardIsShown = false;
+    component.togglePreview();
+    expect(component.mainStoryCardIsShown).toEqual(true);
+  });
+
+  it('should toggle chapter edit options', () => {
+    component.toggleChapterEditOptions(10);
+    expect(component.selectedChapterIndex).toEqual(10);
+    component.toggleChapterEditOptions(10);
+    expect(component.selectedChapterIndex).toEqual(-1);
+  });
+
+  it('should toggle chapter lists', () => {
+    component.chaptersListIsShown = false;
+    component.toggleChapterLists();
+    expect(component.chaptersListIsShown).toEqual(true);
+
+    component.toggleChapterLists();
+    expect(component.chaptersListIsShown).toEqual(false);
+  });
+
+  it('should toggle main story card', () => {
+    component.mainStoryCardIsShown = false;
+    component.toggleStoryEditorCard();
+    expect(component.mainStoryCardIsShown).toEqual(true);
+
+    component.toggleStoryEditorCard();
+    expect(component.mainStoryCardIsShown).toEqual(false);
+  });
+
+  it('should open and close notes editor', () => {
+    component.notesEditorIsShown = false;
+    component.openNotesEditor();
+    expect(component.notesEditorIsShown).toEqual(true);
+
+    component.closeNotesEditor();
+    expect(component.notesEditorIsShown).toEqual(false);
+  });
+
+  it('should return when the node is the initial node', () => {
+    expect(component.isInitialNode('node_1')).toEqual(false);
+    expect(component.isInitialNode('node_2')).toEqual(true);
+  });
+
+  it('should call StoryUpdate to update story title', () => {
+    let storyUpdateSpy = spyOn(storyUpdateService, 'setStoryTitle');
+    component.updateStoryTitle('title99');
+    expect(storyUpdateSpy).toHaveBeenCalled();
+  });
+
+  it('should call StoryUpdate to update story thumbnail filename', () => {
+    let storyUpdateSpy = spyOn(storyUpdateService, 'setThumbnailFilename');
+    component.updateStoryThumbnailFilename('abcd');
+    expect(storyUpdateSpy).toHaveBeenCalled();
```

Comment:
```suggestion
    let storyUpdateSpy = spyOn(storyUpdateService, 'setThumbnailFilename');

    component.updateStoryThumbnailFilename('abcd');

    expect(storyUpdateSpy).toHaveBeenCalled();
```

Please do this for all the remaining tests also

Reply:
done